### ISSUE TITLE
Unable to open workspace in GCC Ops

### ISSUE THREAD

**Issue by anilnatha** [MAAP-Project/ZenHub#429]
As the title states, I'm unable to open workspaces in GCC Ops. The issue occurs with Firefox and Chrome, even when using a private/incognito window.

Steps to reproduce:

1. Open browser and create a new private/incognito window.
2. Visit https://ops.maap-project.org/
3. Login using URS
4. Visit Tools -> Development
5. Create a new workspace using "MAAP Basic Stable"
6. After step 5 begins processing I see some output and then it changes to the second screenshot:

<img width="1280" alt="Screen Shot 2021-08-03 at 13 57 58" src="https://user-images.githubusercontent.com/931107/128085191-7322de4f-4189-4eab-b78a-bfa0d8d6fda9.png">

<img width="1280" alt="Screen Shot 2021-08-03 at 13 58 06" src="https://user-images.githubusercontent.com/931107/128085215-1795b9a5-f008-4495-9c61-297b9fed3eb8.png">

**Comment by anilnatha** [comment]
@bsatoriu Here is a screenshot showing the workspace name that isn't following the workspace naming convention that you pointed out was showing in the logs as well:

<img width="1280" alt="Screen Shot 2021-08-04 at 09 51 36" src="https://user-images.githubusercontent.com/931107/128222794-7cf42c6a-d83a-4c7c-989b-8dd588a1e21f.png">

**Comment by anilnatha** [comment]
@bsatoriu Now that I have updated UAT to point to the UAT ADE environment which is using Eclipse Che 7, it is also exhibiting the same issue with the workspace name, see screenshots below:

<img width="1280" alt="Screen Shot 2021-08-04 at 15 11 25" src="https://user-images.githubusercontent.com/931107/128262196-2341a7be-75f6-4a0a-8d89-cdc56b7340e0.png">

<img width="1280" alt="Screen Shot 2021-08-04 at 15 11 53" src="https://user-images.githubusercontent.com/931107/128262198-de69977a-8725-4104-a48e-4d7887a70bf2.png">

**Comment by gchang** [comment]
That namespace seems messed up. How did you get it to say "anil-natha-work-*'?

**Comment by anilnatha** [comment]
I remember creating my EarthData account, logging into ADE using URS, and after that I vaguely recall the interface putting me on the "Create Custom Workspace" tab and I may have created my first workspace from there. I don't remember clicking on the "Get started" at the time.

**Comment by bsatoriu** [comment]
@anilnatha @gchang the reason behind the workspace startup problem was related to the presence of periods in Anil's URS username, `anil.natha.work`. Che is more strict about allowable username characters than Keycloak, so there were conflicts arising from the mismatch between the username and the Che-generated namespace. On the Che side, I [updated the regex pattern in the namespace validator to allow for hyphens](https://github.com/MAAP-Project/eclipse-che/commit/69c1a282910390b494916fc0d7272d9c90347695) (since these were generated from the invalid "." characters).

Next, in the jupyter-ide, I [updated the entrypoint script to convert any periods in the `CHE_WORKSPACE_NAMESPACE`](https://github.com/MAAP-Project/maap-jupyter-ide/blob/93e85da918e7f8e733cfb2c42ef96e3305baa991/entrypoint.sh) variable to hyphens. 

This fixed is deployed to the DIT environment for testing. I verified this fix there with my own URS account, `brian.test.periods`.

On a side note, Che generates a user's namespace only one time during Che account creation. Consequently, Anil, I'll need to delete your Che accounts on Ops and UAT so your namespace can be regenerated before testing in these environments.

**Comment by oddes** [comment]
@bsatoriu @anilnatha @gchang 
I am able to open workspace but seeing side panel twice as described in the ticket below:
Side panel shows twice on ADE workspace #442

**Comment by bsatoriu** [comment]
@anilnatha @oddes I have removed your test accounts from Ops and UAT ADE environments. This story is ready to test in UAT and Ops.

I also updated the jupyter-ide entrypoint to account for [all disallowed characters](https://github.com/MAAP-Project/maap-jupyter-ide/blob/develop/entrypoint.sh#L14), so any username should now be Che-compatible.

@gchang  The following accounts in Ops have invalid usernames, so I wanted to see if we should just delete them, or maybe also email them to let them know this has been resolved:

ake_ros (Rosenqvist, Ake)
hek_4.mtb (Kay, Heather)
joana.lx.bm (Melo, Joana)

At first I tried checking the postgres db for their generated namespace, before realizing these namespaces are generated in kubernetes (of course):

```
kubectl get secrets --all-namespaces | grep -v che-tls | grep -v che-self | grep -e '-che-'
hayashi-masato-che-2ut0i2       default-token-psh5z                                     kubernetes.io/service-account-token   3      57d
hayashi-masato-che-2ut0i2       che-workspace-token-mdxhk                               kubernetes.io/service-account-token   3      57d
anil-natha-work-che-7ivouq      default-token-2bt69                                     kubernetes.io/service-account-token   3      51d
anil-natha-work-che-7ivouq      che-workspace-token-scsq4                               kubernetes.io/service-account-token   3      51d
hek-4-mtb-che-kboqsn            default-token-7pb9n                                     kubernetes.io/service-account-token   3      18d
hek-4-mtb-che-kboqsn            che-workspace-token-glf2c                               kubernetes.io/service-account-token   3      18d
brian-test-periods-che-8cr3uq   default-token-fwprp                                     kubernetes.io/service-account-token   3      42h
brian-test-periods-che-8cr3uq   che-workspace-token-tbsks                               kubernetes.io/service-account-token   3      42h
test-user-test-che-nfofky       default-token-p9swv                                     kubernetes.io/service-account-token   3      15h
test-user-test-che-nfofky       che-workspace-token-l5trf                               kubernetes.io/service-account-token   3      15h
```

**Comment by anilnatha** [comment]
Thank you @bsatoriu. I will test this today and report back.

Great sleuthing by the way to solve this issue.

**Comment by gchang** [comment]
@bsatoriu I think we can just delete them. It's not like they were able to use it anyway and if they log in the next time, we can reapprove them.

**Comment by anilnatha** [comment]
@bsatoriu I've tested UAT and OPS.

## UAT 👎

I was unable to login and I think this may be due to a separate issue that you were working on based on one of our past conversations.

<img width="1280" alt="Screen Shot 2021-09-03 at 11 34 25" src="https://user-images.githubusercontent.com/931107/132052211-15b66657-04b0-4280-97d1-02087a8fbc10.png">


## OPS 👍

I was able to successfully login, access ADE, and create a workspace and open it.

<img width="1280" alt="Screen Shot 2021-09-03 at 11 37 08" src="https://user-images.githubusercontent.com/931107/132052177-bde89ea0-111a-4bf4-9c58-c8f0ad232c33.png">

**Comment by anilnatha** [comment]
@bsatoriu In my screenshots above, is the "Get user jobs failed" a concern?

**Comment by bsatoriu** [comment]
> @bsatoriu In my screenshots above, is the "Get user jobs failed" a concern?

Yes and no. We shouldn't be generating that error when v1 is released, but since the jobs UI is getting revamped, we may want to simply hide the DPS UI Menu altogether. The other alternative is to fix the jobs list error in Ops. It's a HySDS error that has been a known issue for a while.

**Comment by bsatoriu** [comment]
> @bsatoriu I think we can just delete them. It's not like they were able to use it anyway and if they log in the next time, we can reapprove them.

Good point--I deleted these users. They actually won't need to be approved, since they're already in system, just not the Keycloak database.

**Comment by anilnatha** [comment]
>They actually won't need to be approved, since they're already in system, just not the Keycloak database.

Did y'all receive a notification to approve me in Keycloak after I tried accessing ADE today? Just wanted to verify.

**Comment by bsatoriu** [comment]
No, Anil, the email notifications from Syncope were not sent after your ADE login.

**Comment by bsatoriu** [comment]
@anilnatha I think I'll just redeploy UAT ADE using on Monday since the configuration is out-of-date anyway. I'll put this back as resolved once UAT is ready for testing.

**Comment by anilnatha** [comment]
> No, Anil, the email notifications from Syncope were not sent after your ADE login.

@bsatoriu Is this something we need to open a separate ticket for in case we ever need to delete a user's account in the future and they register again?

**Comment by bsatoriu** [comment]
@anilnatha  no, this is by design. I deleted your keycloak account, not your syncope account. The latter would have triggered an email upon re-registration.
